Grid fixes + saving the window position

AliShug opened the pull request:
This corrects the background grid display at all zoom levels, which was getting misaligned due to integer rounding errors. I also added code to store and restore the main window's position, size and maximised state when closing/opening ModuleSim. Settings are held in user preferences, which is implementation-dependent (on windows it's the user registry)

AliShug committed:
Fixed grid display

Visual only: fixes a problem where the grid was being drawn out of
alignment at most zoom levels. Also makes grid alignment more consistent
at distant zoom.

AliShug committed:
Store window bounds between uses

samtherussell reviewed (commented):
Grid fixes look great. Just one niggle about saving the window position...if you close the program while on a second screen and then tried to open it in a one 1 display set up it's gonna place the window off the screen, which would be rather annoying. Is there are way around that??

AliShug commented:
Yeah I was wondering about that, I'll have to see if there's a way to query the available desktop area(s)

samtherussell commented:
No worries, I've just added the grid stuff in now with a rudimentary help screen I just added.

AliShug committed:
Keep the restored window on-screen

Tested with 2 monitors in extended desktop configuration. The window is
considered "on-screen" if any part of the window intersects any of the
active display areas, which seems to be the standard way of dealing with
this. If the window is off-screen it gets restored to its default size
and centered on the active display (unless it was maximised, in which
case it remains maximised on the visible display)

AliShug commented:
Alright, that should do it 👍 If the window is on any active display it's restored, if not it resets to default size on the current display. Maximise state is preserved either way.

samtherussell commented:
Beautiful. will add it in tomorrow

On 22 Oct 2016 19:23, "Alistair" notifications@github.com wrote:

> Alright, that should do it 👍 If the window is on any active display it's
> restored, if not it resets to default size on the current display. Maximise
> state is preserved either way.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> https://github.com/uobteachingtechnologist/ModuleSim/pull/9#issuecomment-255545101,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/AFNORO685kMZBsNZpB9ANESr_dJhLb1Yks5q2lS0gaJpZM4KdcEU
> .
